### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by EliotAndres**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,37 +39,37 @@ It's also recommended when you have complicated dependencies between your differ
...
 
-In Object Oriented terms, elements would be the Object and entities the Class. 
-Each element belongs to an entity. 
-Each entity defines how elements are created and how they subscribe to other elements.
+In Object Oriented terms, an `element` would be the Object and an `entity` the Class. 
+Each `element` belongs to an `entity`. 
+Each `entity` defines how elements are created and how they subscribe to other `elements`.
 It also has to declare which other entities they depend on. 
 
 ### Source
```
Comment (EliotAndres) [inline README.md]: Use case for source is not very clear

File: README.md [hunk 2]
```diff
@@ -31,37 +39,37 @@ It's also recommended when you have complicated dependencies between your differ
...
 Upon update they can read the state of the elements they are subscribed to and publish their updated state accordingly
 
 ### Entity
 
-In Object Oriented terms, elements would be the Object and entities the Class. 
-Each element belongs to an entity. 
-Each entity defines how elements are created and how they subscribe to other elements.
+In Object Oriented terms, an `element` would be the Object and an `entity` the Class. 
+Each `element` belongs to an `entity`. 
+Each `entity` defines how elements are created and how they subscribe to other `elements`.
```
Comment (EliotAndres) [inline README.md]: I thought elements decided which element to subscribe to

**Comment by EliotAndres** [comment]
Since it seems a bit complex, maybe a draw.io diagram would help understand ?

File: README.md (hunk 2 above)
Comment (arthurandres) [reply README.md]: Yes, I think it should say:
Each `entity` defines how `elements` are created and what element should get notified when a new upstream elements is created (onNewKey).


public interface ElementManager<K, S> {

  ElementUpdater<S> create(K key, UpdateContext updateContext, Toolbox toolbox);

  <K2> void onNewKey(EntityKey<K2, ?> entityKey, K2 key, Consumer<K> toNotify);
}

**Pull request merged**